### PR TITLE
[Merged by Bors] - chore: robustifying for debug.byAsSorry (part 3)

### PR DESCRIPTION
**Pull request by kim-em**
cf #14993 and #15119

### CODE REVIEW

File: Mathlib/CategoryTheory/Category/GaloisConnection.lean [hunk 1]
```diff
@@ -28,8 +28,10 @@ def GaloisConnection.adjunction {l : X → Y} {u : Y → X} (gc : GaloisConnecti
     gc.monotone_l.functor ⊣ gc.monotone_u.functor :=
   CategoryTheory.Adjunction.mkOfHomEquiv
     { homEquiv := fun X Y =>
-        ⟨fun f => CategoryTheory.homOfLE (gc.le_u f.le),
-         fun f => CategoryTheory.homOfLE (gc.l_le f.le), _, _⟩ }
+        { toFun := fun f => CategoryTheory.homOfLE (gc.le_u f.le)
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Category/GaloisConnection.lean:31]: This is a small readability regression to me, but only a small one -> I won't object to landing this.

**Review by grunweg — COMMENTED** [review]
Also looks reasonable, thanks!
maintainer merge

**Comment by grunweg** [comment]
As a meta comment: can you say more *why* is it useful for mathlib to compile under `debug.byAsSorry`? I trust you to have a reason, but would be interested in knowing more about it!

**Comment by grunweg** [comment]
Um, I didn't check CI yet. As it fails, let me revise this to `maintainer delegate`: once you fixed the build, assuming only simple fixes, feel free to maintainer merge on my behalf.

**Comment by semorrison** [comment]
> As a meta comment: can you say more _why_ is it useful for mathlib to compile under `debug.byAsSorry`? I trust you to have a reason, but would be interested in knowing more about it!

Ah! Yes, it means we can test optimizations in Lean much more effectively. Currently it takes ~20 minutes even on a fast machine to see if a change in Lean affects performance deep in Mathlib.

If we can reliably compile Mathlib under `byAsSorry`, then we can maintain a branch of Mathlib with `byAsSorry` turned on everywhere except in a representative sample of slow files. This branch would then give a very fast benchmark.

**Comment by grunweg** [comment]
> > As a meta comment: can you say more _why_ is it useful for mathlib to compile under `debug.byAsSorry`? I trust you to have a reason, but would be interested in knowing more about it!
> 
> Ah! Yes, it means we can test optimizations in Lean much more effectively. Currently it takes ~20 minutes even on a fast machine to see if a change in Lean affects performance deep in Mathlib.
> 
> If we can reliably compile Mathlib under `byAsSorry`, then we can maintain a branch of Mathlib with `byAsSorry` turned on everywhere except in a representative sample of slow files. This branch would then give a very fast benchmark.

This makes a lot of sense - thanks for explaining!

**Review by grunweg — COMMENTED** [review]
Still looks good, and now passes CI -> let me
maintainer merge
again

**Comment by fpvandoorn** [comment]
LGTM

It might also be useful for potential changes to variable inclusion mechanisms (at least if we go back to something similar to Lean 3's method).

bors merge

**Pull request closed without merging**